### PR TITLE
#167271238 Add Validation to Input Form Field

### PR DESCRIPTION
**Pull request by Daymorelah**
#### What does this PR do?
Adds validation to the input form field when an admin adds a new vendor

#### Description of Task to be completed?
* Add custom validation to phone number form field when an admin adds a new vendor
* Change package manager from NPM to Yarn
* Update ReadME file

#### How should this be manually tested?
* Run `yarn build && yarn dev` to start the app
* Login to the app and go to the admin dashboard
* Try to view the list of vendors and try to add a new vendor
* In the phone number form field used to add a new vendor, enter a non-digit and click `Add vendor`
* An error message will show, telling the admin that only digits are required.

#### What are the relevant pivotal tracker stories?
- [#167271238](https://www.pivotaltracker.com/story/show/167271238)

### GIF
add_vendor

### CODE REVIEW

File: client/src/helpers/inputValidation.js [hunk 1]
```diff
@@ -5,6 +5,10 @@ const inputValidation = (formFields) => {
     && key !== 'errors' && key !== 'id') {
       errors[key] = '*required';
     }
+    // eslint-disable-next-line no-useless-escape
+    if (key === 'tel' && /[^0-9\+]/.test(value)) {
+      errors[key] = 'Phone number should contain numbers only';
```
Comment (AmosWels) [inline client/src/helpers/inputValidation.js]: GoodWork @Daymorelah , i see that i works well. How about adding some placeholders also for the field. 

Also i noted that your PR description has some typo for the `field`, you instead have `filed`. Can consider improving it. Can view the image below.

image

Comment (Daymorelah) [reply client/src/helpers/inputValidation.js]: Thank you @AmosWels for your feedback. I have edited my PR description accordingly. However, the input fields already have form labels associated with them adding a place holder might just be overkill. Am just saying.

**Comment by SoniaRMK** [comment]
@Daymorelah thanks for the good work. However, the coverage has dropped. Please look into this.

**Comment by Daymorelah** [comment]
> @Daymorelah thanks for the good work. However, the coverage has dropped. Please look into this.

It happened that no test was written for this feature when it was implementedHowever, I have written tests for the feature and the test coverage has increased.

**Comment by Daymorelah** [comment]
> Great work @Daymorelah, I have tested out the branch and it compiles well. Also, the validation seems to work well. However, the submit button doesn't submit. When clicked, it does nothing.

The `Add Vendor` button works well on my end as seen in the GIF attached in my PR description. If you still have the issue we can synch on it.

**Pull request merged**